src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs

```diff
@@ -0,0 +1,410 @@
+        {
+        }
+
+        public HtmlLogger(IFileHelper fileHelper, IHtmlTransformer htmlTransformer,
+            XmlObjectSerializer dataContractSerializer)
+        {
+            this.fileHelper = fileHelper;
+            this.htmlTransformer = htmlTransformer;
+            this.xmlSerializer = dataContractSerializer;
+        }
+
+        /// <summary>
+        /// Gets the directory under which default html file and test results attachments should be saved.
+        /// </summary>
+        public string TestResultsDirPath { get; private set; }
+
+        /// <summary>
+        /// Total results are stored in sequential order
+        /// </summary>
+        /// <returns></returns>
+        public ConcurrentDictionary<Guid, ObjectModel.TestResult> Results { get; private set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public ConcurrentDictionary<string, TestResultCollection> ResultCollectionDictionary { get; private set; }
+
+        /// <summary>
+        /// Test results stores all the summary and the details of every results in hierarchical order.
+        /// </summary>
+        public TestRunDetails TestRunDetails { get; private set; }
+
+        /// <summary>
+        /// Total passed tests in the test results.
+        /// </summary>
+        public int PassedTests { get; private set; }
+
+        /// <summary>
+        /// Total failed tests in the test results.
+        /// </summary>
+        public int FailedTests { get; private set; }
+
+        /// <summary>
+        /// Total tests in the results.
+        /// </summary>
+        public int TotalTests { get; private set; }
+
+        /// <summary>
+        /// Total skipped tests in the results.
+        /// </summary>
+        public int SkippedTests { get; private set; }
+
+        /// <summary>
+        /// Path to the xml file.
+        /// </summary>
+        public string XmlFilePath { get; private set; }
+
+        /// <summary>
+        /// path to html file.
+        /// </summary>
+        public string HtmlFilePath { get; private set; }
+
+        /// <inheritdoc/>
+        public void Initialize(TestLoggerEvents events, string testResultsDirPath)
+        {
+            if (events == null)
+            {
+                throw new ArgumentNullException(nameof(events));
+            }
+
+            if (string.IsNullOrEmpty(testResultsDirPath))
+            {
+                throw new ArgumentNullException(nameof(testResultsDirPath));
+            }
+
+            // Register for the events.
+            events.TestRunMessage += TestMessageHandler;
+            events.TestResult += TestResultHandler;
+            events.TestRunComplete += TestRunCompleteHandler;
+
+            TestResultsDirPath = testResultsDirPath;
+            TestRunDetails = new TestRunDetails();
+            Results = new ConcurrentDictionary<Guid, ObjectModel.TestResult>();
```

Comment:
>Results [](start = 12, length = 7)

do we need concurrent dictionaries, I'm not sure if multiple threads are trying to add to results.